src/main/java/Duke.java

```diff
@@ -39,15 +42,35 @@ public static void main(String[] args) {
                 break;
             case "list":
                 for (int j = 0; j < counter; j += 1) {
-                    System.out.print((j + 1) + ".[" + tasks[j].getStatusIcon() + "] ");
-                    tasks[j].printDescription();
+                    System.out.println((j + 1) + "." + tasks[j]);
                 }
                 break;
-            default:
-                tasks[counter] = new Task(input);
-                System.out.println("added: " + tasks[counter].description);
+            case "todo":
+                description = input.substring(5);
+                tasks[counter] = new Todo(description);
+                System.out.println("Okay, added:\n" + tasks[counter]);
+                counter += 1;
+                System.out.println("Now you have " + counter + " task(s) in the list.");
+                break;
+            case "deadline":
+                description = input.substring(9, input.indexOf("/by") - 1);
+                by = input.substring(input.indexOf("/by") + 4);
+                tasks[counter] = new Deadline(description, by);
+                System.out.println("Okay, added:\n" + tasks[counter]);
                 counter += 1;
```

Comment:
also consider adding more comments in your code to explain what each part does :)